audio/src/passthrough_decoder.rs

```diff
@@ -0,0 +1,82 @@
+use super::{AudioDecoder, AudioError, AudioPacket};
+use std::error;
+use std::fmt;
+use std::io::{Error, Read, Seek};
+use std::mem;
+use std::slice;
+
+pub struct PassthroughDecoder<R: Read + Seek>(R);
+
+impl<R: Read + Seek> PassthroughDecoder<R> {
+    pub fn new(input: R) -> Result<Self, PassthroughError> {
+        Ok(PassthroughDecoder(input))
+    }
+}
+
+impl<R: Read + Seek> AudioDecoder for PassthroughDecoder<R> {
+    fn seek(&mut self, _: i64) -> Result<(), AudioError> {
+        Err(PassthroughError::SeekNotAllowed)?
+    }
+    fn next_packet(&mut self) -> Result<Option<AudioPacket>, AudioError> {
+        let mut data = [0u8; 8192];
+        let mut rd = 0;
+        while rd < data.len() {
+            let thisrd = self.0.read(&mut data[rd..])?;
+            rd += thisrd;
+            if thisrd == 0 {
+                break;
+            }
+        }
+
+        if rd == 0 {
+            return Ok(None);
+        }
+
+        // This will insert an extraneous 0 byte at EOF if the file size was not an even number of bytes
+        let s16 = mem::size_of::<i16>(); // Yes, I know it is always 2
```

Comment:
What's the point in this if I may ask, given that it'll always be 2?

Reply:
no, not really